lib/Server.js

```diff
@@ -204,6 +204,15 @@ class Server {
       options.allowedHosts = [options.allowedHosts];
     }
 
+    // for cases when it's a singleton array with `all` or `auto`
+    if (
+      Array.isArray(options.allowedHosts) &&
+      options.allowedHosts.length === 1 &&
+      (options.allowedHosts[0] === "auto" || options.allowedHosts[0] === "all")
+    ) {
+      options.allowedHosts = options.allowedHosts[0];
+    }
+
```

Comment:
I think we should also account for `allowedHosts: ['host1.com', 'host2.com', 'all']`:

https://github.com/webpack/webpack-dev-server/blob/8a3f0d8b9d41a4f431ceb4700341ac9e6e585563/lib/Server.js#L1516

```diff
- if (this.options.allowedHosts === "all") { 
+ if (
+ this.options.allowedHosts === "all" || 
+ (Array.isArray(this.options.allowedHosts) && this.options.allowedHosts.includes("all"))
+ ) { 
```

What do you think?

Reply:
`all` and other hosts are mutually exclusive, I don't think there would be a use case for specifying other hosts with all :thinking: Maybe this would be the case for `auto` when we need to allow localhost along with other hosts

Reply:
👍🏻

Reply:
/cc @alexander-akait need slight attention 

is this okay or should we handle 'auto' too here

Reply:
I will look at this in near future

Reply:
will add tests once you confirm

Reply:
will add tests once you confirm

Reply:
I think it was mistake adding `auto` (because we already allow to handle from localhost/ip/etc), we should revisit this in the next major release, I will finish PR